modules/trusted-firmware-m/Kconfig.tfm

```diff
@@ -206,6 +206,19 @@ config TFM_BL2
 	  TFM is designed to run with MCUboot in a certain configuration.
 	  This config adds MCUboot to the build - built via TFM's build system.
 
+config TFM_BUILD_NS
+	bool "Build the TF-M Non-Secure application and libraries"
+	help
+	  When this is enabled, the TF-M build system will build the Non-Secure
+	  application and libraries.
+
+config TFM_APP_NS
+	bool "Use the TF-M Non-Secure application"
+	depends on TFM_BUILD_NS
+	help
+	  When this is enabled, the Zephyr application as a whole will be
+	  replaced with the TF-M Non-Secure application.
```

Comment:
i'm trying to understand the use-case, and trying to avoid settings in global Kconfig if such settings are purely intended for building / running test cases in a single sample.

Maybe for now, keep the setting, but improve the help text.

Also I believe we should overwrite the default found here:
https://github.com/zephyrproject-rtos/zephyr/blob/d5ab3ee0b06457f9e82be12cac1bf39617113e4e/Kconfig.zephyr#L399-L401

so that when `TFM_APP_NS=y` then we will only print memory usage if user specifically requests it.

Reply:
I would prefer to not do it. The memory usage is still right, even though it is swapped the application.
I think  to do what you suggest "correctly" there would be other similar cases (like ram/rom reporting targets) that should be updated as well. I think it is better to just leave it and not start to pollute the rest of the build system with this wierd TF-M configuration.

I've updated the description of TFM_APP_NS so that it should be more clear to users what it does, and that they shouldn't be using that option.

Reply:
> I think it is better to just leave it and not start to pollute the rest of the build system with this wierd TF-M configuration.

when I say overwrite, all you'd need to do would be to add this directly in this file, `Kconfig.tfm`:
```suggestion
 config OUTPUT_PRINT_MEMORY_USAGE 
 	default n if TFM_USE_NS_APP
```

that way we won't print memory usage per default in this particular case.
Of course users can still manually enable it, just like with rom / ram report, but it won't print per default.